codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -906,6 +916,21 @@ public DefaultStream createStream(int streamId) throws Http2Exception {
         }
 
         @Override
+        public DefaultStream createIdleStream(int streamId) throws Http2Exception {
+            return createStream(streamId, IDLE);
+        }
+
+        @Override
+        public DefaultStream createStream(int streamId, boolean halfClosed) throws Http2Exception {
```

Comment:
We had.  This is what I mentioned in my TODO.  The issue is when the stream is created the flow controller would like to know the state the stream will be in to be able to allocate the correct type of object.  If you remember before I was unconditionally creating a `ReducedFlowState` in the `RemoteFlowController` but now I am able to directly create the correct type of object [here](https://github.com/netty/netty/pull/3677/files#diff-d8ca391b616f0c1db6e8f345ca3284feR63) because the state of the stream is known at creation time (and not set at some later time).